tempesta_fw/http.c

```diff
@@ -434,9 +452,13 @@ tfw_http_prep_redirect(TfwHttpMsg *resp, unsigned short status, TfwStr *cookie,
 	 * See RFC 1945 9.3 and RFC 7231 7.1.2.
 	 */
 	if (host.len) {
-		static TfwStr proto = { .ptr = S_HTTP, .len = SLEN(S_HTTP) };
 		ret |= tfw_http_msg_write(&it, resp, &proto);
 		ret |= tfw_http_msg_write(&it, resp, &host);
+		if (req->port.len) {
+			DEFINE_TFW_STR(s_col, ":");
+			ret |= tfw_http_msg_write(&it, resp, &s_col);
+			ret |= tfw_http_msg_write(&it, resp, &req->port);
```

Comment:
Imagine that uri is provided in absolute form (with host information). in this case:
```
req->host == 'example.com'
req->port == '8080'
val(req->h_tbl->tbl[TFW_HTTP_HDR_HOST]) == 'example.com:8080'

=>
host == val(req->h_tbl->tbl[TFW_HTTP_HDR_HOST]) == 'example.com:8080'
```
That means extra `req->port` is added to redirect uri leading to `example.com:8080:8080`.  As for me, `req->host` should contain port subcomponent, and `req->port` wouldn't be needed at all.

Reply:
Please, add an issue to cover all possible cases in functional tests

Reply:
Corrected.